portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayIcons.testcase

```diff
@@ -6,65 +6,95 @@ definition {
-                    layoutName = "Language Test Page Name");
-            }
-
-        task ("Add a Language Selector to page") {
-                JSONLayout.addWidgetToPublicLayout(
-                    groupName = "Test Site Name",
-                    layoutName = "Language Test Page Name",
-                    widgetName = "Language Selector");
-            }
-        }
-
-        tearDown {
-            var testPortalInstance = PropsUtil.get("test.portal.instance");
-
-            if ("${testPortalInstance}" == "true") {
-                PortalInstances.tearDownCP();
-            }
-            else {
-                JSONGroup.deleteGroupByName(groupName = "Test Site Name");
-            }
-        }
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+		task ("Add a site") {
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
+
+		task ("Add a widget page") {
+			JSONLayout.addPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Language Test Page Name");
+		}
+
+		task ("Add a Language Selector to page") {
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Language Test Page Name",
+				widgetName = "Language Selector");
+		}
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			JSONGroup.deleteGroupByName(groupName = "Test Site Name");
+		}
+	}
+
+	/** JONH'S REVIEW:
+	-DONE.  
+	Use all capitals for path locator titles  <td>FLAG_enUS</td>
+
+
+	- The path locator pattern for en-us and ca-es are very similar. Why not combine them into 1 locator like //*[*[name()='svg'][contains(@class,'${key_languageID}')]]	
+	- Also, as we talked about before, I think it would be better to target the href attribute b/c that's the attribute that ensures the proper flag icon is displayed on the page based on experimenting with the DOM.
+	So something like this helps to better make sure that Spain flag is present: //*[name()='svg']//*[contains(@href,'ca-es')].
+
+
+	- task ("When click on localization button") {
+			Click(locator1 = "Icon#FLAG_enUS");
+	I think existing path LANGUAGE#DROPDOWN path would work fine here.
+
+
+	- DONE: Moved to When task.
+	Assertions are not needed in Given task.
+		AssertElementPresent(locator1 = "Icon#FLAG_enUS");
+
+			LanguagePortlet.viewCurrentLanguage(locale = "en-US");
+
+
+	- task ("Then flag icons are present on the button and dropdown menu items") {
+	The task says we're asserting flag icons meaning multiple icons but this task is only asserting 1 icon.
+	Also, were you able to try out ocular to see if it would better meet this need for all icons rendering ok?
+	If we're not using ocular, I think something along the lines of LanguagePortlet.viewLanguages macro under Icon and else if Icon Menu section would document better what we're trying to assert
+
+	**/
```

Comment:
I'm not comfortable about leaving this trail of review history in these intermediate commits.

Reply:
Yes, sure, you're totally right!  I didn't realize that it was saved in a commit, it was just the notes I needed when working on it but I thought they were just saved locally and I deleted them before making the commit.
Git is a snitch! 

Sorry for the inconveniences and thanks for letting me know, I will take into account for the future.